plugin.rb

```diff
@@ -521,4 +521,40 @@ def self.enqueue_assign_hooks(event, payload)
     end
   end
 
+  register_search_advanced_filter(/in:assigned/) do |posts|
+    if @guardian.can_assign?
+      posts.where("topics.id IN (
+        SELECT tc.topic_id
+        FROM topic_custom_fields tc
+        WHERE tc.name = 'assigned_to_id' AND
+                        tc.value IS NOT NULL
+        )")
+    end
+  end
+
+  register_search_advanced_filter(/in:not_assigned/) do |posts|
```

Comment:
Can we make this `in:unassigned`. Then it will match the rest of the UI better:

![Screenshot 2020-09-09 at 15 47 40](https://user-images.githubusercontent.com/6270921/92614331-d67ff800-f2b3-11ea-999c-a6ec81475758.png)


(Sorry, I know this means changing it in a few places 😓 )